holidays/holiday_base.py

```diff
@@ -959,6 +959,32 @@ def get_named(
 
         raise AttributeError(f"Unknown lookup type: {lookup}")
 
+    def get_entries_sorted(self) -> dict[date, str]:
+        return {k: v for k, v in sorted(self.items(), key=lambda item: item[0])}
+
+    def get_next_holiday(
+        self, start: DateLike = None, previous: bool = False
+    ) -> Union[tuple[date, str], tuple[None, None]]:
+        """Return the date and name of the next holiday from provided date
+        (if previous is False) or the previous holiday (if previous is True).
+        If no date is given the search starts from current date"""
+
+        dt = self.__keytransform__(start) if start else datetime.now().date()
+        if not previous:
+            next_date = next((x for x in self.get_entries_sorted() if x > dt), None)
+            if not next_date and dt.year < self.end_year:
+                self._populate(dt.year + 1)
+                next_date = next((x for x in self.get_entries_sorted() if x > dt), None)
+        else:
+            next_date = next((x for x in reversed(self.get_entries_sorted()) if x < dt), None)
+            if not next_date and dt.year > self.start_year:
+                self._populate(dt.year - 1)
+                next_date = next((x for x in reversed(self.get_entries_sorted()) if x < dt), None)
```

Comment:
```suggestion
        if not previous:
            next_date = next((x for x in sorted(self.keys()) if x > dt), None)
            if not next_date and dt.year < self.end_year:
                self._populate(dt.year + 1)
                next_date = next((x for x in sorted(self.keys()) if x > dt), None)
        else:
            next_date = next((x for x in sorted(self.keys(), reverse=True) if x < dt), None)
            if not next_date and dt.year > self.start_year:
                self._populate(dt.year - 1)
                next_date = next((x for x in sorted(self.keys(), reverse=True) if x < dt), None)
```
It looks faster and doesn't involve additional functions. And don't squash changes into single commit every time, please.

Reply:
OK - added your change.